### PR TITLE
Android Nested RecyclerView

### LINKED ISSUE #1411: Android Nested RecyclerView
### Brief Summary: 
Android studio provides developers with an easy way to implement a recycler view. This guide will discuss and implement a nested recycler view. A nested recycler view is a recycler view inside another recycler view.

### Key Takeaways: 
One of the perfect examples that showcase the nested recycler view is the Google play store. 

NestedRecyclerView1

In this case, the app integrates several recycler views. Nas, a child to one main recycler view.
Each recycler view will then be used to display distinct data. Google PLAY STORE nested recycler view is scrollable horizontally.

Recycler views are diverse, and they and best when it comes to displaying different data into your application. You can modify the view to meet your data representation.
This guide will discuss and implement an application that showcases a nested recycler view 

### References: 
https://developer.android.com/guide/topics/ui/layout/recyclerview

### PR DESCRIPTION
**Pull request by kimkimani**
### Article Checklist
**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process, quality, and style-once published, you can then submit any number of articles.)**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

 fixes #1411

### CODE REVIEW

**Comment by hectorkambow** [comment]
@paulodhiambo is looking to review this article.

File: articles/android-nested-recycler-view/index.md [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+    app:layout_constraintEnd_toEndOf="parent"
+    app:layout_constraintHorizontal_bias="0.01"
+    app:layout_constraintStart_toStartOf="parent"
+    app:layout_constraintTop_toBottomOf="@+id/Movie_category" />
+</androidx.constraintlayout.widget.ConstraintLayout>
+```
+
+#### Child RecyclerView layout items
+
+We have designed the parent View and the items that belong to it, including the child RecyclerView. We need to add the items to the child RecyclerView. These are the items that will be scrollable horizontally in each child RecyclerView widgets that we add. Create an XML file, name it `child_recyclerview_items.xml`, and include the layout below.
```
Comment (paulodhiambo) [inline articles/android-nested-recycler-view/index.md:105]: Consider changing the statement "... RecyclerView widgets that we add" to "... RecyclerView widget that we add."

File: articles/android-nested-recycler-view/index.md [hunk 2]
```diff
@@ -0,0 +1,400 @@
...
+}
+```
+
+### Setting up Adapter classes
+
+Adapters are used to set and pass data that will be displayed in the respective RecyclerView items. The adapter manages the collection of items (defined in an individual layout) and `ViewHolder` objects. `ViewHolder` defines the individual views (widgets) in the layout and populate the components into an individual row layout. An `onBindViewHolder` class will bind the views with their respective data objects.
+
+Technically, the data could come from a database or a remote server if the information to be passed to the recycling view is hosted remotely. To keep this app straight to the point, we will use dummy/locally available information and then pass them to the respective RecyclerView Adapters.
+
+Let populate this data into the respective views. We will have two view adapters for each of the RecyclerView widgets. Create the two adapters.
```
Comment (paulodhiambo) [inline articles/android-nested-recycler-view/index.md:193]: Consider changing "Let populate this data into the respective views." to "Let us populate this data into the respective views."

File: articles/android-nested-recycler-view/index.md [hunk 3]
```diff
@@ -0,0 +1,400 @@
...
+        ChildRecyclerViewAdapter childRecyclerViewAdapter = new ChildRecyclerViewAdapter(arrayList,holder.childRecyclerView.getContext());
+            holder.childRecyclerView.setAdapter(childRecyclerViewAdapter);
+        }
+}
+```
+
+As you can see, we are binding the information of the  `ChildRecyclerViewAdapter` to the `ParentRecyclerViewAdapter`. I have set different data categories. In this case, each category contains its distinct data. The `ArrayList` of all the categories binds to the parent RecyclerView. In the `onBindViewHolder`, we are setting up the child RecyclerView by
+
+- Initializing and setting the [`LinearLayoutManager`](https://developer.android.com/reference/androidx/recyclerview/widget/LinearLayoutManager) (create an instance of the layout manager) for the child RecyclerView. Here we specify that this view will be `HORIZONTAL`.
+- Invoke the layout fixed size. This tells the layout not to move around the screen. It stabilizes the layout screen movements.
```
Comment (paulodhiambo) [inline articles/android-nested-recycler-view/index.md:357]: This statement is not clear, consider explaining.

File: articles/android-nested-recycler-view/index.md [hunk 4]
```diff
@@ -0,0 +1,400 @@
...
+- Pass `ArrayList` of the child items to the respective `ParentViewHolder` as `category.setText(currentItem.movieCategory())` of the parent RecyclerView. Each category will inflate with respective array list items consisting of a movie name and a hero image. Make sure to check the content of this project on [GitHub](https://github.com/kimkimani/Nested_RecyclerView) to get all the necessary image resources for each `movieCategory`.
+- Associate the child Adapter and the RecyclerView. This informs the RecyclerView which adapter it should work with.
+
+Let set the necessary categories to display the movies to the main RecyclerView layout.
```
Comment (paulodhiambo) [inline articles/android-nested-recycler-view/index.md:361]: Consider rewriting "Let set the necessary ..." to "Let us set the necessary ..."

**Review by paulodhiambo — CHANGES_REQUESTED** [review]
@kimkimani amazing tutorial. I have mentioned a few suggestions, review those whenever you get a chance. I found a few grammatical errors. Use Grammarly to ensure minimal errors. That way, the article gets reviewed faster.

**Comment by paulodhiambo** [comment]
@hectorkambow this article needs plagiarism check.

**Review by paulodhiambo — APPROVED** [review]

**Comment by paulodhiambo** [comment]
@hectorkambow technical review done. ready for final review

**Pull request merged**